app/gui/src/project-view/components/visualizations/TableVisualization.vue

```diff
@@ -391,7 +316,7 @@ function addRowIndex(data: object[]): object[] {
   return data.map((row, i) => ({ [INDEX_FIELD_NAME]: i, ...row }))
 }
 
-function toField(name: string, valueType?: ValueType | null | undefined): ColDef {
+function toField(name: string, index?: number, valueType?: ValueType | null | undefined): ColDef {
```

Comment:
Because this function has multiple optional parameters, passing an options object containing the optional values as fields could make its call sites easier to read.
The parameters could also use some documentation: It's not obvious when `index` is expected to be present or what happens if its absent.